### ISSUE TITLE
Working on a library making shamir's secret sharing easier to use

### ISSUE THREAD

**Issue by simonfrey** [posener/sharedsecret#10]
I am contacting you here, as I did not find any twitter/email contact of yours:

I am currently working on bringing Shamir's secret sharing as easy to use web app into the browser as I found there is no GUI for it (as far as Wikipedia knows).

As I am not a cryptographer myself I would love your feedback on the project and if there are errors I made that render the crypto/secret sharing wrong.

s4 (Simple Shamir's Secret Sharing) is implemented with golang and than used in a vue.js frontend via webassembly. The project can be found on https://github.com/simonfrey/s4 and the ready to use frontend is deployed on my page: simon-frey.com/s4/

In manner of "never do roll your own crypto" the crypto is based on two libraries:
-  Yours for the Shamir's Secret Sharing algorithm
- Cryptopasta for the AES usage of golang AES

Navigation in the repo:
- The shamir's secret sharing is done in the top level file `s4.go`
- The webassembly combining shamir's secret sharing and AES can be found in `frontend/wasm/wasm.go`
- All the (IMO for the crypto not important) javascript FU can found in `frontend/js`

Thank you for your help and expertise. I hope we can build a usable GUI for Shamir's secret sharing together in order to bring this amazing math to more people :D

**Comment by simonfrey** [comment]
Similiar message I posted to reddit [/r/crypto](https://www.reddit.com/r/crypto/comments/h9eqi1/simple_shamirs_secret_sharing_in_the_browser_s4/)

**Comment by posener** [comment]
Hey there!
This is a very nice project. Thanks for sharing!
Sadly I'm not a security expert, just a humble programmer :-)
Cheers!

**Comment by smiklosovic** [comment]
Hi @posener  and @simonfrey,

are these solutions done over finite GF fields? The wiki says that integer approach is not too much secure.

Check the problem and solution section:

https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing

**Comment by posener** [comment]
This library uses `math.Big` not integer.
Not sure what it means regarding "finite fields".
Does that answer the question?

**Comment by smiklosovic** [comment]
This is the implementation over GF(256) https://github.com/codahale/shamir

**Comment by posener** [comment]
Again, I *think* that the `math.Big` is acceptable fot that matter, as it can be an arbitrary large number. But I assume that you should have an extra expert set of eyes 👀.

**Comment by smiklosovic** [comment]
It does not matter how "big" these numbers are imho ... read the wiki article about that here

https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing

Read the section "Problem" which starts with this sentence: 

Although the simplified version of the method demonstrated above, which uses integer arithmetic rather than finite field arithmetic, works fine, there is a security problem: Eve gains a lot of information about {\displaystyle S}S with every {\displaystyle D_{i}}D_{i} that she finds.

Basically, as soon as there is an integer artithmetic involved instead of Galois fields modulo irreducible polynomial, it is in theory possible to crack it if you know some information.

**Comment by simonfrey** [comment]
FYI: I actually changed to the hashicorps shamir algorithm a while back, just as they are a big corp and their implementation is used widely in the industry with vault: https://github.com/hashicorp/vault/blob/main/shamir/shamir.go

They use GF(256)

**Comment by posener** [comment]
Interesting.
From what I see from the code, each share is a uint8, no?
But fair enough. If it is widely used it seems more reasonable to be used by your app.
Thanks for letting me know, and thanks for this conversation! Sorry I could not provide you with a good answer.

**Comment by posener** [comment]
It seems that this conversation was ended
